services/statestorage/layered_state_test.go

```diff
@@ -0,0 +1,203 @@
+package statestorage
+
+import (
+	"fmt"
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-network-go/services/statestorage/adapter"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestWriteAtHeight(t *testing.T) {
+	persistenceMock := &StatePersistenceMock{}
+	persistenceMock.
+		When("Read", primitives.ContractName("c"), "k1").
+		Return((*protocol.StateRecord)(nil), false, nil).
+		Times(1)
+	d := newDriver(persistenceMock, 5)
+	d.write(1, "c", "k1", "v1")
+
+	v, exists, err := d.read(0, "c", "k1")
```

Comment:
your naming makes it hard to understand what the test does. Try this here:
`_, exists, err := d.read(0, "c", "k1")`